src/foam/dao/XMLDAO.java

```diff
@@ -0,0 +1,76 @@
+package foam.dao;
+
+import foam.core.FObject;
+import foam.core.XMLSupport;
+
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import javax.xml.stream.XMLStreamException;
+import foam.core.X;
+import foam.core.ProxyX;
+
+public class XMLDAO
+        extends MapDAO
+{
+
+    private String fileName;
+
+    public void setFileName(String filename) {
+        if (filename.contains(".xml")){
+            fileName = System.getProperty("user.dir") + filename;
+        } else {
+            fileName = System.getProperty("user.dir") + filename.concat(".xml");
+        }
+    }
+
+    public String getFileName() { return fileName; }
+
+    // Read file and read data in the DAO
+    public void init() throws IOException {
+        X x = new ProxyX();
+        this.setX(x);
+
+        try {
+            XMLSupport.XMLToFObject(fileName, this);
+        } catch (IllegalAccessException | XMLStreamException | FileNotFoundException ex) {
+        }
+    }
+
+    // Rewrites file when new object is put into DAO
+    public FObject put(FObject obj) {
+        this.setOf(obj.getClassInfo());
```

Comment:
Add a property to DAO called 'saveEnabled' or similar. Set it to false when loading from the XML file to prevent feedback. Don't rewrite XMLFile if set.